contrib/babelfishpg_common/src/varbinary.c

```diff
@@ -164,10 +164,12 @@ varbinaryin(PG_FUNCTION_ARGS)
 	int			len;
 	bytea	   *result;
 	int32 typmod = PG_GETARG_INT32(2);
+	const char *dump_restore = GetConfigOption("babelfishpg_tsql.dump_restore", true, false);
 
 	len = strlen(inputText);
 
-	if (typmod == TSQLHexConstTypmod)
+	if (typmod == TSQLHexConstTypmod ||
+		(dump_restore && strcmp(dump_restore, "on") == 0)) /* Treat input string as T-SQL hex constant during restore */
```

Comment:
is it safe to do this decoding unconditionally during restore? what if default is not hex encoded string?

Reply:
Yeah, pg_dump always dumps binary/varbinary values as hex encoded string, so unconditional decoding is safe here. Secondly, no matter what the default value was, dump will always contain hex encoded string since varbinary output function is written like that.